datafusion/optimizer/src/utils.rs

```diff
@@ -92,6 +94,21 @@ pub fn log_plan(description: &str, plan: &LogicalPlan) {
     trace!("{description}::\n{}\n", plan.display_indent_schema());
 }
 
+/// check whether the expression is volatile predicates
+pub(crate) fn is_volatile_expression(e: &Expr) -> Result<bool> {
+    let mut is_volatile_expr = false;
+    e.apply(&mut |expr| {
+        Ok(if is_volatile(expr)? {
+            is_volatile_expr = true;
+            VisitRecursion::Stop
+        } else {
+            VisitRecursion::Continue
+        })
+    })
+    .unwrap();
```

Comment:
should we use the `unwrap`? or the `?`

Reply:
because the result of `is_volatile_expression` is `result`

Reply:
Thanks for pointing this out. I forgot to change it when I moved the code.